### PR TITLE
add types to completion code

### PR DESCRIPTION
**Pull request by devoncarew**
- add additional types to the completion code
- run our tests in checked mode

@lukechurch

### CODE REVIEW

File: lib/src/common_server.dart [hunk 1]
```diff
@@ -13,7 +13,7 @@ import 'package:rpc/rpc.dart';
 import 'analyzer.dart';
 import 'compiler.dart';
 
-import 'completer_driver.dart' as completer;
+import 'completer_driver.dart' as completer_driver;
```
Comment (devoncarew) [inline lib/src/common_server.dart]: renamed this, as seeing variables named `completer` definitely make me think that the object was an instance of the `Completer` class.

File: lib/src/completer_driver.dart [hunk 2]
```diff
@@ -31,8 +31,8 @@ typedef void NotificationProcessor(String event, params);
 Stream<bool> analysisComplete;
 StreamController<bool> _onServerStatus;
 
-Stream completionResults;
-StreamController _onCompletionResults;
+Stream<Map> completionResults;
```
Comment (devoncarew) [inline lib/src/completer_driver.dart]: Added types, to help me trace through the code.

File: lib/src/common_server.dart [hunk 3]
```diff
@@ -45,11 +45,10 @@ class CompileResponse {
 }
 
 class CompleteResponse {
-  String result;
+  final String result;
 
-  CompleteResponse(List<String> completions) {
+  CompleteResponse(Iterable<String> completions) :
```
Comment (devoncarew) [inline lib/src/common_server.dart]: The call to this ctor would have failed if the server was running in check mode (List was expected, but an iterable was passed in). I changed our tests to run in check mode to help catch this. We should also have an integration test for the completion service.

Comment (lukechurch) [inline lib/src/common_server.dart]: We do have, just as discussed they aren't integrated yet. I suspect they are however in unchecked mode, I'll see and fix that

File: lib/src/common_server.dart (hunk 1 above)
Comment (lukechurch) [inline lib/src/common_server.dart]: sg.

**Comment by lukechurch** [comment]
lgtm, thanks!

**Pull request merged**